lib/circuitry/queue_creator.rb

```diff
@@ -0,0 +1,52 @@
+require 'circuitry/services/sqs'
+require 'circuitry/queue'
+
+module Circuitry
+  class QueueCreator
+    include Services::SQS
+
+    attr_reader :queue_name
+
+    def self.find_or_create(queue_name, dead_letter_queue_name: nil, max_receive_count: 8 )
+      creator = new(queue_name)
+      result = creator.create_queue
+      creator.create_dead_letter_queue(dead_letter_queue_name, max_receive_count) if dead_letter_queue_name
+      result
+    end
+
+    def create_queue
+      @_queue = Queue.new(create_primary_queue_internal) unless defined?(@_queue)
+      @_queue
+    end
+
+    def create_dead_letter_queue(name, max_receive_count)
+      @_dl_queue = Queue.new(create_dl_queue_internal(name, max_receive_count)) unless defined?(@_dl_queue)
+      @_dl_queue
+    end
+
+    def initialize(queue_name)
```

Comment:
nitpick, but I think `initialize` should be the first instance method in any class.